src/OrbitGl/TimelineUiTest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "GlCanvas.h"
+#include "MockBatcher.h"
+#include "MockTextRenderer.h"
+#include "MockTimelineInfo.h"
+#include "TimelineUi.h"
+#include "Viewport.h"
+
+namespace orbit_gl {
+
+class TimelineUiTest : public TimelineUi {
+ public:
+  TimelineUiTest(int width)
+      : TimelineUi(nullptr /*parent*/, &mock_timeline_info, &viewport_, &layout_),
+        primitive_assembler_(&mock_batcher_),
+        viewport_(width, 0) {
+    mock_timeline_info.SetWorldWidth(width);
+  }
+
+  void TestUpdatePrimitives(uint64_t min_tick, uint64_t max_tick, bool is_small_screen = false) {
+    mock_text_renderer_.Clear();
+    mock_batcher_.ResetElements();
+    mock_timeline_info.SetMinMax(min_tick, max_tick);
+
+    UpdatePrimitives(primitive_assembler_, mock_text_renderer_, min_tick, max_tick,
+                     PickingMode::kNone);
+
+    int num_labels = mock_text_renderer_.GetNumAddTextCalls();
+    int num_boxes = mock_batcher_.GetNumBoxes();
+    int num_major_ticks = mock_batcher_.GetNumLinesByColor(kMajorTickColor);
+    int num_minor_ticks = mock_batcher_.GetNumLinesByColor(kMinorTickColor);
+
+    // Lines should be only major and minor ticks.
+    EXPECT_EQ(mock_batcher_.GetNumLines(), num_major_ticks + num_minor_ticks);
+
+    // Major ticks should always be between 2 and 10, given scale set. In extremely small screens
+    // (Width < 160), we don't force a minimum number of major ticks.
+    if (!is_small_screen) EXPECT_GE(num_major_ticks, 2);
```

Comment:
This should have braces: 

```if (!is_small_screen) {
  EXPECT_GE(num_major_ticks, 2);
}```

Reply:
Technically this is fine (which is not meant to discourage the braces)
https://google.github.io/styleguide/cppguide.html#Conditionals